app/javascript/components/Profile.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { Fragment } from 'react';
+import { Divider, Menu } from 'semantic-ui-react';
+import NavBar from './Navigation/NavBar';
+import { inject, observer } from 'mobx-react';
+import SessionStore from '../stores/session_store';
+import Footer from './Navigation/Footer';
+import { RouteComponentProps } from 'react-router';
+import Account from './Profile/Account';
+import ChangePassword from './Profile/ChangePassword';
+import { RouterStore } from 'mobx-react-router';
+import UserStore from '../stores/user_store';
+import UserList from './Profile/UserList';
+import DeleteAccount from './Profile/DeleteAccount';
+
+interface MatchParams {
+  part: string;
+}
+
+interface ProfileProps extends RouteComponentProps<MatchParams> { }
+
+interface InjectedProps extends ProfileProps {
+  sessionStore: SessionStore;
+  routerStore: RouterStore;
+  userStore: UserStore;
+}
+
+@inject('sessionStore', 'routerStore', 'userStore')
+@observer
+export default class Profile extends React.Component<ProfileProps> {
+  get injected() {
+    return this.props as InjectedProps;
+  }
+
+  get part() {
```

Comment:
why defining a getter instead of assigning this value to a const variable inside render?